src/tag.c

```diff
@@ -3740,14 +3741,21 @@ jumpto_tag(
 
     // copy the command to pbuf[], remove trailing CR/NL
     str = tagp.command;
-    for (pbuf_end = pbuf; *str && *str != '\n' && *str != '\r'; )
+    if (VIM_ISDIGIT(*str))
+    {
+	// need to inject a ':' for a proper :nr command
+	isdigit = TRUE;
+	vim_snprintf((char *)pbuf, LSIZE, ":");
```

Comment:
As this is copying only one character, can you this be replaced with `pbuf[0] = ':'; pbuf[1] = NUL;`

Reply:
makes sense, will change